pkg/cmd/pulumi/state_move.go

```diff
@@ -141,6 +142,9 @@ func (cmd *stateMoveCmd) Run(
 	for _, res := range sourceSnapshot.Resources {
 		matchedArg := resourceMatches(res, args)
 		if matchedArg != "" {
+			if strings.HasPrefix(string(res.Type), "pulumi:providers:") {
+				return errors.New("cannot move providers")
```

Comment:
Is this going to be presented to the user directly? If so perhaps we should make this a little more descriptive (is there a sensible "why" we can communicate?), or indicate what they can do instead?